server/http/routes/dataQueries.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { Request, Response } from 'express';
+import * as H from '../../utils/helpers';
+import * as LOG from '../../utils/logger';
+import * as DBAPI from '../../db';
+
+import GraphQLApi from '../../graphql';
+import { GetSystemObjectDetailsInput, GetSystemObjectDetailsResult } from '../../types/graphql';
+
+export async function routeRequest(request: Request, response: Response): Promise<void> {
+
+    const detailsToReturn = request.params.id;
+    console.warn(detailsToReturn+'|'+JSON.stringify(request.params));
+
+    // if nothing then complain
+    if(detailsToReturn===undefined) {
+        LOG.error('HTTP request: incorrect usage of endpoint', LOG.LS.eHTTP);
+        response.send('Request failed. Incorrect use of endpoint. Be sure to include what you are looking for');
+        return;
+    }
+
+    // handle the proper type
+    switch(detailsToReturn){
+        case 'systemObject': {
+            return await getSystemObjectDetails(request,response);
+        } break;
+
+        default: {
+            LOG.error(`HTTP request: unsupported request (${detailsToReturn})`, LOG.LS.eHTTP);
+            response.send(`Request failed. Unsupported request/path (${detailsToReturn})`);
+        }
+    }
+}
+
+// convenience routine routine for getting system object details to be used with routes.
+// NOTE: not connected as it should not be 'live' until an API is created and protected
+async function getSystemObjectDetails(req: Request, response: Response): Promise<void> {
+
+    // grab our config options from query params
+    const subjectLimit: number = (req.query.limit)?parseInt(req.query.limit as string):10000;
+    const systemObjectId: number = (req.query.objectId)?parseInt(req.query.objectId as string):-1;
+
+    // fetch all subjects from Packrat DB to get list of IDs
+    const systemObjects: DBAPI.SystemObject[] | null = await DBAPI.SystemObject.fetchAll(); /* istanbul ignore if */
+    if (!systemObjects) {
+        sendResponseMessage(response,false,'could not get system objects from DB');
+        return;
+    }
+    if(systemObjects.length<=0) {
+        sendResponseMessage(response,false,'no system objects found in DB');
+        return;
+    }
+    LOG.info(`Getting SystemObject Details processing ${systemObjects.length} ids`,LOG.LS.eGQL);
+
+    // loop through subjects, extract name, query from EDAN
+    const output: string[] = [];
+    for(let i=0; i<systemObjects.length; i++) {
+
+        if(i>=subjectLimit) break;
+
+        const idSystemObject: number = systemObjects[i].fetchID();
+        if(systemObjectId>0 && idSystemObject!=systemObjectId) continue;
+
+        const input: GetSystemObjectDetailsInput = {
+            idSystemObject
+        };
+        const graphQLApi = new GraphQLApi(true);

```

Comment:
This is an interesting approach ... and I suspect it is likely heavier weighted that what you intend.  Instead of routing this request through the graphQL machinery, why not call getSystemObjectDetails() directly?